File: solutions/cpp/grains/1/grains.cpp
#include "grains.h"

namespace grains {

    uint64_t square(uint8_t grid) {
        return (uint64_t)1 << (grid - 1);
    }
    
    uint64_t total() {
        return ~(uint64_t)0;
    }

}  // namespace grains

File: solutions/csharp/grains/1/Grains.cs
public static class Grains
{
    public static ulong Square(int n)
    {
        if (n < 1 || n > 64)
        {
            throw new ArgumentOutOfRangeException("n must be between 1 and 64");
        }
        return 1UL << (n - 1);
    }

    public static ulong Total() => ~0UL;
}